### PR TITLE
module_utils - Deprecation of sort_json_policy_dict (Fix sanity issue)

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

Deprecation of `sort_json_policy_dict()` function to solve CI sanity issues

##### ISSUE TYPE

- Bugfix Pull Request

### CODE REVIEW

**Review by alinabuzachis — APPROVED** [review]

**Comment by tremble** [comment]
regate

File: changelogs/fragments/20240411-remove-deprecated-sort_json_policy_dict.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+deprecated_features:
+  - module_utils.policy - deprecation of the function ``sort_json_policy_dict()`` (https://github.com/ansible-collections/amazon.aws/pull/2052).
```
Comment (tremble) [inline changelogs/fragments/20240411-remove-deprecated-sort_json_policy_dict.yml:3]: This isn't a deprecation, this is a removal.  The deprecation has already happened.

**Review by tremble — CHANGES_REQUESTED** [review]
This was actually already part of #2040

File: changelogs/fragments/20240411-remove-deprecated-sort_json_policy_dict.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+removed_features:
+  - module_utils.policy - deprecation of the function ``sort_json_policy_dict()`` (https://github.com/ansible-collections/amazon.aws/pull/2052).
```
Comment (tremble) [inline changelogs/fragments/20240411-remove-deprecated-sort_json_policy_dict.yml:3]:
```suggestion
  - module_utils.policy -the previously deprecated ``sort_json_policy_dict()`` function has been removed, consider using ``compare_policies()`` instead (https://github.com/ansible-collections/amazon.aws/pull/2052).
```

**Review by tremble — APPROVED** [review]

**Comment by abikouo** [comment]
recheck

**Pull request merged**